ci/dockerfiles/base-ci-linux/Dockerfile

```diff
@@ -0,0 +1,136 @@
+FROM docker.io/library/ubuntu:22.04
+
+ARG VCS_REF=master
+ARG BUILD_DATE=""
+ARG REGISTRY_PATH=docker.io/analoglabs
+
+# metadata
+LABEL summary="Layer 1 image with all dependencies for Rust compilation." \
+      name="${REGISTRY_PATH}/base-ci-linux" \
+      maintainer="lohann@analog.one" \
+      version="1.0.0" \
+      description="libssl-dev, clang, libclang-dev, libsasl2-dev, lld, cmake, make, git, pkg-config \
+curl, jq, time, lsof, rhash, rust stable, rust nightly, sccache, cargo-udeps, cargo-tarpaulin" \
+      one.analog.image.vendor="Analog.one" \
+      one.analog.image.source="https://github.com/Analog-Labs/chain-connectors/blob/${VCS_REF}/\
+ci/dockerfiles/base-ci-linux/Dockerfile" \
+      one.analog.image.documentation="https://github.com/Analog-Labs/chain-connectors/blob/${VCS_REF}/\
+ci/dockerfiles/base-ci-linux/README.md" \
+      one.analog.image.revision="${VCS_REF}" \
+      one.analog.image.created="${BUILD_DATE}"
```

Comment:
This is the base image I'm using to build everything, it is based on Parity's official images, with some improvements I did:
- Use clang-15 instead of clang-14
- Support all architectures 🔥  (x86_64, aarch64, armv7, etc), the original polkadot image only support x86_64, which is sad.
- Implicit version for all cargo packages, this way if we build this image in 2 years, we will get reliable results.
- Automatically detect the architecture and add the correct `linux-musl` rust target

References:
- https://github.com/paritytech/scripts/blob/master/dockerfiles/base-ci-linux/Dockerfile
- https://github.com/paritytech/scripts/blob/master/dockerfiles/ci-linux/Dockerfile
- https://github.com/paritytech/polkadot/blob/master/scripts/ci/dockerfiles/polkadot/polkadot_builder.Dockerfile